env/file_system_tracer.cc

```diff
@@ -338,6 +338,23 @@ IOStatus FSRandomAccessFileTracingWrapper::InvalidateCache(size_t offset,
   return s;
 }
 
+IOStatus FSRandomAccessFileTracingWrapper::ReadAsync(
+    const IOOptions& opts, IODebugContext* dbg, FSReadRequest* req,
+    std::function<void(FSReadResponse* resp)> cb, IOHandle* io_handle) {
+  StopWatchNano timer(clock_);
+  timer.Start();
+  IOStatus s = target()->ReadAsync(opts, dbg, req, cb, io_handle);
+  uint64_t elapsed = timer.ElapsedNanos();
+  uint64_t io_op_data = 0;
```

Comment:
I am unsure what this implementation is going to accomplish.  If the read is truly asynchronous, this trace will tell us only how long the read took to schedule, not how long it took to complete, when it completed, or its completion status.

Reply:
Removed the API from the PR.